### PR TITLE
Add a connection listener that emits telemetry events

### PR DESCRIPTION
**Pull request by v0idpwn**
Alternative to #310. Would have to write documentation, etc.

### CODE REVIEW

**Review by josevalim — COMMENTED** [review]
I like this! We need to improve the docs to say exactly how to use and the events emitted but it works for me!

**Comment by v0idpwn** [comment]
I added two documentation sections. I'm afraid it's a bit hidden in the documentation, but the `DBConnection` module doc is already quite big :)

File: lib/db_connection/telemetry_listener.ex [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  ## Usage
+
+  Start the listener, optionally using a name, and pass it under the
+  `:connection_listeners` option when starting DbConnection:
+
+      {:ok, pid} = TelemetryListener.start_link([name: MyListener])
+      {:ok, _conn} = DBConnection.start_link(SomeModule, [connection_listeners: [MyListener]])
+      # Using a tag, which will be sent in telemetry metadata
+      {:ok, _conn} = DBConnection.start_link(SomeModule, [connection_listeners: {[pid], :my_tag}])
+
```
Comment (josevalim) [inline lib/db_connection/telemetry_listener.ex:17]: What do you think about showing an example with a supervision tree perhaps?

**Review by josevalim — APPROVED** [review]
One tiny comment and ship it!

**Comment by v0idpwn** [comment]
Almost merged it as `DbConnection` instead of `DBConnection` 😄

**Comment by josevalim** [comment]
Done!

**Pull request merged**